src/Middleware/Diagnostics/test/testassets/ExceptionHandlerSample/StartupWithProblemDetails.cs

```diff
@@ -17,12 +17,22 @@ public void ConfigureServices(IServiceCollection services)
     public void Configure(IApplicationBuilder app)
     {
         // Configure the error handler to produces a ProblemDetails.
-        app.UseExceptionHandler();
+        app.UseExceptionHandler(new ExceptionHandlerOptions
```

Comment:
Technically this makes the other test in ExceptionHandlderTest.cs not test the default behavior. Could we change it to something like
```csharp
app.Map("/throw", throwApp =>
{
    throwApp.UseExceptionHandler();
    throwApp.Map("/conflict", throwConflictApp =>
    {
        throwConflictApp.Run(_ => throw new Exception("Conflict Exception"));
    });
    throwApp.Run(_ => throw new Exception("Application Exception"));
});

app.Map("/throw2", throwApp =>
{
    throwApp.UseExceptionHandler(new ExceptionHandlerOptions
    {
        StatusCodeSelector = ex => ex is ConflictException
            ? StatusCodes.Status409Conflict
            : StatusCodes.Status500InternalServerError,
    });
    throwApp.Map("/conflict", throwConflictApp =>
    {
        throwConflictApp.Run(_ => throw new Exception("Conflict Exception"));
    });
});
```

So the tests can use different usages of `UseExceptionHandler`?

Reply:
Thanks for pointing out, improved the isolation of the test as suggested.